internal/controller/resources/authpolicy.go

```diff
@@ -214,3 +225,74 @@ func (k *kserveAuthPolicyTemplateLoader) getGatewayInfo(ctx context.Context, log
 
 	return gateways
 }
+
+type clientAuthPolicyStore struct {
+	client client.Client
+}
+
+func NewClientAuthPolicyStore(client client.Client) AuthPolicyStore {
+	return &clientAuthPolicyStore{
+		client: client,
+	}
+}
+
+func (c *clientAuthPolicyStore) Get(ctx context.Context, key types.NamespacedName) (*unstructured.Unstructured, error) {
+	authPolicy := &unstructured.Unstructured{}
+	c.setAuthPolicyGVK(authPolicy)
+
+	err := c.client.Get(ctx, key, authPolicy)
+	if err != nil {
+		return nil, fmt.Errorf("could not GET AuthPolicy %s: %w", key, err)
+	}
+	return authPolicy, nil
+}
+
+func (c *clientAuthPolicyStore) Remove(ctx context.Context, key types.NamespacedName) error {
+	authPolicy := &unstructured.Unstructured{}
+	c.setAuthPolicyGVK(authPolicy)
+	authPolicy.SetName(key.Name)
+	authPolicy.SetNamespace(key.Namespace)
+
+	if err := c.client.Delete(ctx, authPolicy); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("could not DELETE AuthPolicy %s: %w", key, err)
+	}
+	return nil
+}
+
+func (c *clientAuthPolicyStore) Create(ctx context.Context, authPolicy *unstructured.Unstructured) error {
+	c.setAuthPolicyGVK(authPolicy)
+
+	if err := c.client.Create(ctx, authPolicy); err != nil {
+		return fmt.Errorf("could not CREATE AuthPolicy %s/%s: %w", authPolicy.GetNamespace(), authPolicy.GetName(), err)
+	}
+	return nil
+}
+
+func (c *clientAuthPolicyStore) Update(ctx context.Context, authPolicy *unstructured.Unstructured) error {
+	c.setAuthPolicyGVK(authPolicy)
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		current := &unstructured.Unstructured{}
+		c.setAuthPolicyGVK(current)
+		if err := c.client.Get(ctx, types.NamespacedName{
+			Name:      authPolicy.GetName(),
+			Namespace: authPolicy.GetNamespace(),
+		}, current); err != nil {
+			return err
+		}
+
+		authPolicy.SetResourceVersion(current.GetResourceVersion())
+		return c.client.Update(ctx, authPolicy)
+	})
+}
+
+func (c *clientAuthPolicyStore) setAuthPolicyGVK(authPolicy *unstructured.Unstructured) {
```

Comment:
Does it need to be a receiver function?

Reply:
Right, it does not need to be a receiver function. I took it out and updated